sklearn/decomposition/tests/test_pca.py

```diff
@@ -403,45 +416,58 @@ def test_pca_score(svd_solver):
     ll2 = pca.score(rng.randn(n, p) * 0.2 + np.array([3, 4, 5]))
     assert ll1 > ll2
 
-    pca = PCA(n_components=2, whiten=True, svd_solver=svd_solver)
+    pca = PCA(n_components=2, whiten=True, svd_solver=svd_solver, random_state=rng)
     pca.fit(X)
     ll2 = pca.score(X)
     assert ll1 > ll2
 
 
-def test_pca_score3():
+def test_pca_score3(global_random_seed):
```

Comment:
Just including the `global_random_seed` fixture made the test highly seed sensitive, as for 24 out of 100 seeds, the test score was highest for the ("wrong") model with two components.
I believe the test is supposed to address the following point: A possible strategy to select an appropriate number of components is to compare the likelihood for held-out data, e.g. in a cross-validation scheme as illustrated in [this example](https://scikit-learn.org/stable/auto_examples/decomposition/plot_pca_vs_fa_model_selection.html#sphx-glr-auto-examples-decomposition-plot-pca-vs-fa-model-selection-py). However, we cannot expect the test-likelihood of the "true" model to be maximal for every (isolated) pair of train and test datasets, but only on average. So I changed the test to run 10 trials with different train-test pairs for each `n_components` and accumulate the test scores.
Too keep the test fast, I also reduced the sample size, as the original sample size of 200 required 15 trials to make the test seed-insensitive in the 0-99 range and with 50 samples only 10 trials are required.